src/gui/base/ColumnEmptyMessageBox.ts

```diff
@@ -8,47 +8,59 @@ import type { lazy } from "@tutao/tutanota-utils"
 import { assertMainOrNode } from "../../api/common/Env"
 
 assertMainOrNode()
+
+// If you change this make sure you pass through all the attrs
 export type Attrs = {
 	message: TranslationKey | lazy<Children>
 	icon?: AllIcons
 	color: string
 }
-/**
- * A message box displaying a text. A message box can be displayed on the background of a column if the column is empty.
- */
 
-export default class ColumnEmptyMessageBox implements Component<Attrs> {
+/** Displays a big message with an option icon above it. */
+export class IconMessageBox implements Component<Attrs> {
 	view({ attrs }: Vnode<Attrs>): Children {
-		return m(
-			".fill-absolute.flex.col.items-center.justify-center",
+		return m(".flex.col.items-center.justify-center.mlr", [
+			attrs.icon
+				? m(Icon, {
+						icon: attrs.icon,
+						style: {
+							fill: attrs.color,
+						},
+						class: "icon-message-box",
+				  })
+				: null,
 			m(
-				".flex.col.items-center.justify-center.mlr",
+				".h2.text-center.text-preline",
 				{
 					style: {
-						"margin-top": px(attrs.icon ? -size.icon_message_box - size.vpad_xl : -size.vpad_xl),
+						color: attrs.color,
 					},
 				},
-				[
-					attrs.icon
-						? m(Icon, {
-								icon: attrs.icon,
-								style: {
-									fill: attrs.color,
-								},
-								class: "icon-message-box",
-						  })
-						: null,
-					m(
-						".h2.text-center.text-preline",
-						{
-							style: {
-								color: attrs.color,
-							},
-						},
-						getMessage(attrs),
-					),
-				],
+				getMessage(attrs),
 			),
+		])
+	}
+}
+
+/**
+ * A message displaying a text. A message box can be displayed on the background of a column if the column is empty. The text inside of it will be centered vertically, taking the icon into account.
+ */
+
+export default class ColumnEmptyMessageBox implements Component<Attrs> {
+	view({ attrs }: Vnode<Attrs>): Children {
+		return m(
+			".fill-absolute.flex.col.items-center.justify-center",
+			{
+				style: {
+					"margin-top": px(attrs.icon ? -size.icon_message_box - size.vpad_xl : -size.vpad_xl),
+				},
+			},
+			// If we pass plain attrs all lifecycle callbacks we attach from the outside will be called twice, once on the wrong element.
+			m(IconMessageBox, {
```

Comment:
does {...attrs} not work here?

Reply:
By the comment above not really

Reply:
right, was thinking we'd be adding handlers later, but they might already be attached